src/components/screens/management-screen.tsx

```diff
@@ -0,0 +1,438 @@
+  const [ displayedPersonsTotalTime, setDisplayedPersonsTotalTime ] = React.useState<PersonWithTotalTime[]>([]);
+  const [ selectedPersonWithTotalTime, setSelectedPersonWithTotalTime ] = React.useState<PersonWithTotalTime | undefined>(undefined);
+  const [ searchInput, setSearchInput ] = React.useState("");
+  const context = React.useContext(ErrorContext);
+  const history = useHistory();
+
+  /**
+   * Populate one person's total work data
+   * 
+   * @param person person with total time data
+   */
+  const populatePersonTotalTimeData = async (person: PersonWithTotalTime) => {
+    let personTotalTime: PersonWithTotalTime = {
+      ...person
+    };
+
+    try {
+      await Api.getTimeBankApi()
+        .timebankControllerGetTotal({
+          personId: personTotalTime.person.id.toString(),
+          retention: TimebankControllerGetTotalRetentionEnum.ALLTIME
+        })
+        .then(totalTime => {
+          personTotalTime = {
+            ...personTotalTime,
+            timeEntryTotal: totalTime[0]
+          };
+        });
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchTimeDataFailed, error);
+    }
+    return personTotalTime;
+  };
+
+  /**
+   * Fetches the person data & person total time data
+   */
+  const fetchData = async () => {
+    const fetchedPersons: PersonWithTotalTime[] = [];
+
+    setIsLoading(true);
+
+    try {
+      await Api.getTimeBankApi()
+        .timebankControllerGetPersons()
+        .then(persons => {
+          persons.forEach(person =>
+            fetchedPersons.push({
+              person: person
+            } as PersonWithTotalTime));
+        });
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchUserDataFailed, error);
+    }
+
+    const personsTotalTimePromises = fetchedPersons.map(populatePersonTotalTimeData);
+
+    const fetchedPersonsTotalTime = await Promise.all(personsTotalTimePromises);
+
+    setPersonsTotalTime(fetchedPersonsTotalTime);
+    setDisplayedPersonsTotalTime(fetchedPersonsTotalTime);
+    setIsLoading(false);
+  };
+
+  React.useEffect(() => {
+    if (!accessToken) {
+      return;
+    }
+
+    const decodedToken: any = jwt_decode(accessToken.access_token);
+    const roles = decodedToken.realm_access.roles as string[];
+
+    if (!roles.includes("admin")) {
+      history.push("/");
+    }
+
+    fetchData();
+  }, []);
+
+  /**
+   * Person Redirect click handler
+   * 
+   * @param person person data
+   */
+  const handlePersonRedirectClick = (person: PersonDto) => {
+    dispatch(setPerson(person));
+    history.push("/");
+  };
+
+  /**
+   * List Item click handler
+   * 
+   * @param event mouse event
+   * @param personWithTotalTime person with total time data
+   */
+  const handleListItemClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>, personWithTotalTime: PersonWithTotalTime) => {
+    setSelectedPersonWithTotalTime(personWithTotalTime);
+  };
```

Comment:
It is possible to add multiple arrow functions to single const. First function takes the normal parameters and the second one handles the caller parameters. In this case the event is not needed so = () => can be blank. IF you'd need to access the event you would simply add the event parameter inside the second ().
```suggestion
  /**
   * List Item click handler
   * 
   * @param personWithTotalTime person with total time data
   */
  const handleListItemClick = (personWithTotalTime: PersonWithTotalTime) => () => {
    setSelectedPersonWithTotalTime(personWithTotalTime);
  };
```